wormhole-connect/src/views/v2/Bridge/ReviewTransaction/GasSlider.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { makeStyles } from 'tss-react/mui';
+import { useDebounce } from 'use-debounce';
+
+import { useTheme } from '@mui/material';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import Collapse from '@mui/material/Collapse';
+import Slider, { SliderThumb } from '@mui/material/Slider';
+import Stack from '@mui/material/Stack';
+import Switch from '@mui/material/Switch';
+import { styled } from '@mui/material/styles';
+import Typography from '@mui/material/Typography';
+
+import config from 'config';
+import TokenIcon from 'icons/TokenIcons';
+import { getDisplayName, calculateUSDPrice } from 'utils';
+import { RootState } from 'store';
+import { setToNativeToken } from 'store/relay';
+
+import { toFixedDecimals } from 'utils/balance';
+import { TokenConfig } from 'config/types';
+
+const useStyles = makeStyles()(() => ({
+  card: {
+    width: '100%',
+    cursor: 'pointer',
+    maxWidth: '420px',
+    overflow: 'visible',
+    padding: '0 4px',
+  },
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '8px',
+    width: '100%',
+  },
+  amounts: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+}));
+
+type SliderProps = {
+  baseColor: string;
+  railColor: string;
+};
+
+const StyledSlider = styled(Slider)<SliderProps>(
+  ({ baseColor, railColor }) => ({
+    color: baseColor,
+    height: 8,
+    '& .MuiSlider-rail': {
+      height: '8px',
+      backgroundColor: railColor,
+      opacity: '10%',
+    },
+    '& .MuiSlider-track': {
+      height: '8px',
+    },
+    '& .MuiSlider-thumb': {
+      height: 20,
+      width: 20,
+      backgroundColor: '#C1BBF6',
+    },
+  }),
+);
+
+const StyledSwitch = styled(Switch)((props) => ({
+  padding: '9px 12px',
+  right: `-12px`, // reposition towards right to negate switch padding
+  '& .MuiSwitch-switchBase.Mui-checked': {
+    color: '#C1BBF6',
+  },
+  '& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track': {
+    backgroundColor: '#C1BBF6',
+  },
+  '& .MuiSwitch-track': {
+    height: '20px',
+    borderRadius: '9px',
+  },
+}));
+
+const GasSlider = (props: { destinationGasFee: number; disabled: boolean }) => {
+  const { classes } = useStyles();
+  const dispatch = useDispatch();
+  const theme = useTheme();
+
+  const { token: sourceToken, toChain: destChain } = useSelector(
+    (state: RootState) => state.transferInput,
+  );
+
+  const {
+    usdPrices: { data },
+  } = useSelector((state: RootState) => state.tokenPrices);
+
+  const prices = data || {};
+
+  const destChainConfig = config.chains[destChain!];
+  const sourceTokenConfig = config.tokens[sourceToken];
+  const nativeGasToken = config.tokens[destChainConfig!.gasToken];
```

Comment:
I don't prefer using the `!` operator to force TS compiler for prop's existence. Ideally the code should be able to handle the existence of `destChainConfig` at runtime.